src/graph.rs

```diff
@@ -1926,19 +2125,83 @@ impl<'a, 'graph> Builder<'a, 'graph> {
       .entry(specifier.clone())
       .or_default()
       .insert(maybe_assert_type);
-    if !self.graph.module_slots.contains_key(specifier) {
-      self
-        .graph
-        .module_slots
-        .insert(specifier.clone(), ModuleSlot::Pending);
-      let specifier = specifier.clone();
-      let maybe_range = maybe_range.map(ToOwned::to_owned);
-      let fut = self
-        .loader
-        .load(&specifier, is_dynamic)
-        .map(move |res| (specifier, maybe_range, res));
-      self.pending.push(Box::pin(fut));
+    if self.graph.module_slots.contains_key(specifier) {
+      return;
+    }
+
+    let maybe_range = maybe_range.map(ToOwned::to_owned);
+    if let Some(npm_resolver) = &self.npm_resolver {
+      match specifier.scheme() {
+        "npm" => {
+          match NpmPackageReqReference::from_specifier(specifier) {
+            Ok(package_ref) => {
+              if self
+                .requested_npm_registry_info_loads
+                .insert(package_ref.req.name.clone())
+              {
+                // request to load
+                let package_name = package_ref.req.name.clone();
+                let fut =
+                  npm_resolver.load_and_cache_npm_package_info(&package_name);
+                self
+                  .pending_npm_registry_info_loads
+                  .push(Box::pin(async move { (package_name, fut.await) }));
+              }
+              self.pending_npm_specifiers.push((
+                specifier.clone(),
+                package_ref,
+                maybe_range,
+              ));
+            }
+            Err(err) => {
+              self.graph.module_slots.insert(
+                specifier.clone(),
+                ModuleSlot::Err(ModuleGraphError::LoadingErr(
+                  specifier.clone(),
+                  maybe_range,
+                  Arc::new(err.into()),
+                )),
+              );
+            }
+          }
+          return;
+        }
+        "node" if npm_resolver.supports_node_specifiers() => {
+          let name = specifier.path();
+          let module_slot = if npm_resolver.is_builtin_node_module(name) {
+            self.graph.has_node_specifier = true;
+            ModuleSlot::Module(Module::External(ExternalModule {
+              specifier: specifier.clone(),
+            }))
```

Comment:
In CLI I'm facing a problem with it - the actual URL pointing to an existing ES module that implements any Node.js built-in module is something like: `internal:ext/node/polyfills/http.ts`. If I return that URL then somehow it gets loaded by the CLI file fetcher, which obviously can't find it. 

To circumvent it I just return here something like `node:http` (which is essentially what was requested). 

What's the reason for external modules to have a different specifier than the one requested? Maybe we should add `Module::BuiltIn` variant that wouldn't remap the specifier to anything else?

Reply:
Added `Module::Node`